### PR TITLE
Add styled system

### PR DESCRIPTION
**Pull request by benadam11**
Instead of adding our own theme api - I thought it would be wise to adopt a well defined spec. 

https://styled-system.com/

Its not fully fleshed out but here is the theme spec in particular https://styled-system.com/theme-specification

Here is a great article detailing SS - https://varun.ca/styled-system/

### CODE REVIEW

File: docs/src/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { Fragment, useState } from "react";
 import { render } from "react-dom";
 
 // TODO: Fix workspace link
-import { Button, Grid } from "../../packages/web";
+import { Button, Grid, Base } from "../../packages/web";
```
Comment (raphamorim) [inline docs/src/index.js:5]: What `Base` does?

File: packages/web/components/Primitives/Base.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import { StyledBase } from "./Styled";
+import { useTheme } from "../../../themes";
+
+export function Base(props) {
+  const theme = useTheme();
+  return <StyledBase theme={theme} m={2} {...props} />;
```
Comment (raphamorim) [inline packages/web/components/Primitives/Base.js]: What `m` prop means?

File: packages/web/components/Primitives/Styled.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  position,
+  zIndex,
+  top,
+  right,
+  bottom,
+  left,
+  // Misc
+  buttonStyle,
+  variant
+} from "styled-system";
```
Comment (raphamorim) [inline packages/web/components/Primitives/Styled.js:77]: I liked the concept, it's going to work with linaria too right?

File: docs/src/index.js (hunk 1 above)
Comment (benadam11) [reply docs/src/index.js:5]: Base is a Styled component that gets all of the helpers from `styled-system` applied to it ... My thought was I want there to be constraints on most elements on their api (what they can and can't do) but to have one generic component that can do everything

File: packages/web/components/Primitives/Base.js (hunk 2 above)
Comment (benadam11) [reply packages/web/components/Primitives/Base.js]: Margin - https://styled-system.com/table

File: packages/web/components/Primitives/Styled.js (hunk 3 above)
Comment (benadam11) [reply packages/web/components/Primitives/Styled.js:77]: Yeah! Works with any Styled component like library

Comment (benadam11) [reply packages/web/components/Primitives/Styled.js:77]: once we fix the Parcel plugin for linaria we can move back to Linaria

File: packages/web/components/Primitives/Base.js (hunk 2 above)
Comment (raphamorim) [reply packages/web/components/Primitives/Base.js]: 🤔

Comment (benadam11) [reply packages/web/components/Primitives/Base.js]: `margin` also works (m is just shorthand)

**Review by raphamorim — APPROVED** [review]

**Pull request merged**